doc/compiler/runtime/JITHooks.md

```diff
@@ -0,0 +1,89 @@
+<!--
+Copyright (c) 2020, 2020 IBM Corp. and others
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+or the Apache License, Version 2.0 which accompanies this distribution and
+is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# Background
+
+The VM has a mechansim to register callbacks for specific runtime event.
+The JIT makes use of these events to perform tasks either for functional 
+correctness, performance, or housekeeping. 
+
+# Hooks
+This section outlines various events or Hooks the JIT registers a callback
+against. The callbacks associated with these Hooks can be found in
+HookedByTheJit.cpp.
+
+|Hook|Description|Comment|
+|--|--|--|
+|`J9HOOK_VM_PROFILING_BYTECODE_BUFFER_FULL`|Notifies when the bytecode profiling buffer (consumed by the IProfiler) for a given thread is full.||
+|`J9HOOK_VM_LOOKUP_JNI_ID`|Notifies when a method or field ID is looked up via JNI.|The JIT uses this to get notified when `main` in the Java code is about to be run; it then unregisters this hook.|
+|`J9HOOK_VM_INITIALIZE_SEND_TARGET`|Notifies when J9Methods are to have their send target initialized.|Used to set initial invocation counts.|
+|`J9HOOK_MM_OMR_LOCAL_GC_START`|Notifies when a local Garbage Collection has started.|Local here applies to policies such as GenCon, where only the nursery region is being GC'd.|
+|`J9HOOK_MM_OMR_LOCAL_GC_END`|Notifies when a local GC has ended|If Real Time GC is enabled, the JIT registers `jitHookLocalGCEnd`; otherwise it registers `jitHookReleaseCodeLocalGCEnd`.|
+|`J9HOOK_MM_OMR_GLOBAL_GC_START`|Notifies when a global GC has started.|Global here implies a GC of the entire heap.|
+|`J9HOOK_MM_OMR_GLOBAL_GC_END`|Notifies when a global GC has ended.|If Real Time GC is enabled, the JIT registers `jitHookGlobalGCEnd`; otherwise it registers `jitHookReleaseCodeGlobalGCEnd`.|
+|`J9HOOK_MM_OMR_GC_CYCLE_END`|Notifies when the GC Cycle has ended.|The JIT only registers against this hook if Real Time GC is enabled.|
+|`J9HOOK_VM_INTERNAL_CLASS_LOAD`|Notifies when a class has been loaded.||
+|`J9HOOK_VM_CLASS_PREINITIALIZE`|Notifies prior to notify the initialization of a class prior to running the `<clinit>`.|
```

Comment:
This phrasing is a bit cumbersome. Could we make it clearer with something like: `Notifies prior to initialization of a class through the <clinit> method`

Reply:
Ah good catch, the phrasing was different before, and I just did a mass search/replace which clearly didn't work out in this case.

Reply:
Fixed in https://github.com/eclipse/openj9/pull/10205/commits/42c8adf0e62227dce6137f78649803fb61249ef1